pkg/apis/mattermost/v1alpha1/clusterinstallation_types.go

```diff
@@ -68,6 +71,16 @@ type ClusterInstallationSpec struct {
 	IngressAnnotations map[string]string `json:"ingressAnnotations,omitempty"`
 }
 
+// Canary defines the configuration of Canary deployment for a ClusterInstallation
+type Canary struct {
+	// Enable defines if Canary deployment will be applied.
+	// +optional
+	Enable bool `json:"enable,omitempty"`
+	// CanaryDeployment defines the canary deployment.
+	// +optional
+	CanaryDeployment AppDeployment `json:"canaryDeployment,omitempty"`
```

Comment:
Can we just name this `Deployment`? `CanaryDeployment` is a bit redundant

Reply:
Thanks for the suggestion. Fixed ;)